### ISSUE TITLE
`bundle update` API is confusing

### ISSUE THREAD

**Issue by vfonic** [bundler/bundler#6990]
Hi!

Let's say there's a gem I'm using in my rails app, let's call it "Slugger". Slugger (`2.0.0`) depends on another gem, let's call it "Mugger" (`spec.add_dependency 'mugger', '~> 7.0.0'`). Mugger just released a new version (`7.1.0`) and it seems to have introduced some changes that break my app.
Slugger also released a new version (`2.1.0`).

Now I want to update Slugger to the latest version, but NOT update Mugger. Slugger didn't change its dependency version specification for Mugger (`spec.add_dependency 'mugger', '~> 7.0.0'`), hence old Mugger version should/would work just fine.

So, I tried:
1. `bundle update slugger` => mugger got updated
(`git reset --hard HEAD`)
2. `bundle update --source slugger` => mugger got updated
(`git reset --hard HEAD`)
3. `bundle update --conservative slugger` => mugger got updated

NOTE: I do have the newer version of mugger locally on my machine, but I think it would be an unexpected behavior if bundler would use it for that reason, especially in the case of `--conservative`.

What am I doing wrong?

I think `bundle update` API is confusing/obscure. I'm doing ruby/rails development for almost 10 years now and I still don't know how `bundle update` works, besides the "default" `bundle update` with no extra arguments.
I checked the documentation, but it's still not clear:
https://bundler.io/man/bundle-update.1.html (not sure why the page has `.1.` in its name)

* `--source=<name>` - does something completely different from what I need (I saw this option on stackoverflow: https://stackoverflow.com/questions/11263150/update-just-one-gem-with-bundler)
* `--conservative`
> Use bundle install conservative update behavior and do not allow shared dependencies to be updated.

Hmmmm, but it didn't work. Am I misunderstanding something?

Also, this documentation page is slightly misleading: https://bundler.io/v2.0/guides/updating_gems.html#updating-a-gem-without-modifying-the-gemfile
> It will, however, update dependencies of other gems if necessary. For instance, if the latest version of `rack-cache` specifies a dependency on `rack >= 1.2.2`, bundler will update `rack` to `1.2.2` even though you have not asked bundler to update `rack`.

I think it would be clearer if it was written that "if there's newer version available, it would get updated, no matter that the existing version is compatible". For example, app uses `rack` `1.2.2` and if there's `rack` `1.2.3`, after bundle update, the `rack` gem will get updated".

> If bundler needs to update a gem that another gem depends on, it will let you know after the update has completed.

I don't think this is the case anymore. Maybe it's just unclear wording, but I don't understand what exactly does this say that bundler does?

I can try to create a repo that will reproduce this issue. Can you please confirm that what I'm seeing is unexpected behavior?

Thanks!

**Comment by deivid-rodriguez** [comment]
@vfonic I agree with your expectations. It looks like a bug in the `--conservative` option to me.

Regarding all the confusing docs, how about creating a PR to improve them?

**Comment by vfonic** [comment]
@deivid-rodriguez I'm not sure I understand 100% what's the desired functionality. The reason I opened this issue was to kick off a discussion about possibly changing the functionality.

I think that the ideal solution would be:
`bundle update` - updates all gems (keeping in mind any dependency gems version restrictions)
`bundle update <gem_name>` - update the gem and all of its dependencies
`bundle update <gem_name> --conservative` - update the bare minimum of the gem and its dependencies. For example, if the dependency requirements/restrictions haven't changed, don't update dependencies.

**Comment by vfonic** [comment]
As far as I know, bundler is a sponsored project. Is this correct?

Perhaps some of the development time could go into documentation improvements? I think having an incorrect/misleading documentation should be quite high priority for fixing.
